Remove constexpr square root call in CDI

alexrlongne opened the pull request:
### Background

* I used `constexpr` square root in a `Require` function to make sure overflow would not result when an input value was squared. I thought this was a good idea because the `sqrt(std::numeric_limits<double>::max)` was previously being computed every time through this loop. Unfortunately, it seems that debug code versions were not evaluating this expression at compile time and it was causing very slow runtimes and even more problems under profiling (the `ce_sqrt` is not an efficient version of `sqrt` it's just one that can be marked `constexpr`!). This PR removes the ce_sqrt call and replaces it with the sqrt of the max double. There is a comment explaining the magic number.

### Purpose of Pull Request

* Fix slow runtimes and profiling times in debug mode

### Description of changes

* Replace constexpr squar root call with a numeric value

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [ ] Travis/Appveyor CI checks pass
  * [ ] Code coverage does not decrease
  * [ ] [Valgrind test passes](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [ ] [Toss3 checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [ ] [Trinitite checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

KineticTheory commented:
Something seems wrong with the history of this branch.  The diff shows a bunch of files from `kde` as modified.  Can you rebase and/or squash?

alexrlongne committed:
Use a numeric value instead of constexpr square root function in CDI

+ Change Require to use numeric value instead of ce_sqrt value, which
was slow in some circumstances

alexrlongne commented:
@KineticTheory Sorry about that, it should be fixed now.
